Test/git-issues/git-issue-1587.dfy

```diff
@@ -0,0 +1,4 @@
+// RUN: %dafny_0 /compile:0 "%s" > "%t"
+// RUN: %diff "%s.expect" "%t"
+
+datatype Foo = Foo(Keys: ())
```

Comment:
Could we add tests for the other cases as well, since we not cover IsLimit, IsSucc, etc?